airbyte-integrations/connectors/source-talkdesk-explore/source_talkdesk_explore/streams.py

```diff
@@ -24,15 +24,19 @@ class GenerateReportStream(HttpStream):
 
     primary_key = None
 
-    def __init__(self, base_path, start_date, timezone, **kwargs):
+    def __init__(self, base_path, start_date, timezone, region, **kwargs):
         super().__init__(**kwargs)
         self.base_path = base_path
         self.start_date = start_date
         self.timezone = timezone
+        self.region = region
 
     @property
-    def url_base(self) -> str:
-        return "https://api.talkdeskapp.com/data/"
+    def url_base(self, **kwargs) -> str:
+        # Can we remove **kwargs here?
+        region_dict = {"US": ".com", "Europe": ".eu", "Canada": "ca.com"}
+        region_domain = region_dict.get(self.region)
+        return f"https://api.talkdeskapp{region_domain}/data/"
```

Comment:
I would suggest to create a constant outside the class
```python
MAP_REGION_DOMAIN = {"US": ".com"...}
```

I think this source can be refactored. Create a BaseStream(HttpStream) where only overwrite the url_base method and make both classes inherent from the BaseStream. With this you don't duplicate code.

Reply:
Done here: https://github.com/airbytehq/airbyte/pull/17587/commits/93ba88ffb4fa74c1f02cf46f61f912f67ed675ba
hopefully it looks better let me know!